src/github.com/couchbase/sync_gateway/rest/server_context.go

```diff
@@ -724,12 +724,22 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(config *DbConfig, useExisti
 		revCacheSize = db.KDefaultRevisionCacheCapacity
 	}
 
+	unsupportedOptions := &db.UnsupportedOptions{}
+	if config.Unsupported != nil {
+		if config.Unsupported.UserViews != nil {
+			if config.Unsupported.UserViews.Enabled != nil {
+				unsupportedOptions.EnableUserViews = *config.Unsupported.UserViews.Enabled
+			}
+		}
+	}
+
```

Comment:
We're using pointers to structs in the config (overall) for a couple of reasons:
1. So that omitempty works during JSON marshalling, and doesn't return an empty struct
2. To differentiate between cases where the config property isn't defined, and when it's defined but equal to the zero value for the property, so we know when to revert to default values.
The second one is more of an issue for things like int values than for structs, but the first one applies here.